document/static/js/modules/editor/state_plugins/table_menu.js

```diff
@@ -0,0 +1,96 @@
+import {Schema} from "prosemirror-model"
+import { Plugin, PluginKey} from "prosemirror-state"
+import {TableMenuDialog} from '../dialogs'
+
+const key = new PluginKey('tableMenu')
+const doc = {content: 'tableMenu'},
+    tableMenu = {
+        content: 'inline*',
+        parseDOM: [{table: ''}],
+        toDOM() {
+            return ["div", {
+                class: ''
+            }, 0]
+        }
+    },
+    text = {group: 'inline'}
+
+const schema = new Schema({
+    nodes: {doc, tableMenu, text},
+    marks: {}
+})
+
+
+export class TableView {
+    constructor(node, view, getPos, options) {
+        this.node = node
+        this.view = view
+        this.getPos = getPos
+        this.options = this.options
+        this.dom = document.createElement("div")
+        this.dom.classList.add(`table-${node.attrs.width}`,`table-${node.attrs.aligned}`,'container');
+        let menu_btn = document.createElement("button")
+        menu_btn.className = "menu-btn"
+        menu_btn.classList.add('btn-disabled')
+        let div = document.createElement("div")
+        div.className = 'menu-stripe'
+        menu_btn.append(div)
+        this.dom.appendChild(menu_btn)
+        this.dom.lastElementChild.addEventListener('click', event => {
+            event.preventDefault()
+            const dialog = new TableMenuDialog(node, view, options)
+            dialog.init() 
+        })
+        let table = document.createElement("table")
+        let tbody = document.createElement("tbody")
+        table.append(tbody)
+        this.contentDOM = this.dom.appendChild(table)
+        this.contentDOM.classList.add(`layout-${node.attrs.layout}`)
+        this.dom.appendChild(this.contentDOM)
+    }
+
+    stopEvent(event) {
+        return true
+    }
+
+    ignoreMutation(record) {
+        return true
+    }
+
+    update(node){
+        let table = findTable(this.view.state)
```

Comment:
I think using findTable() here is running a bit too much code because this is being run every time anything changes, also unrelated to the node. If you look at the documentation for nodeViews [1], you can see that there is a special `selectNode()` and `deselectNode()` method that can you should be able to add and remove the 'btn-disabled' class. Something like
```
selectNode() {
    this.dom.querySelector('.menu-btn').classList.remove('btn-disabled')
}

deselectNode() {
    this.dom.querySelector('.menu-btn').classList.add('btn-disabled')
}
```


[1] https://prosemirror.net/docs/ref/#view.NodeView